### PR TITLE
Reduce size of index build meta in Etcd

### PR DESCRIPTION
**Pull request by jiangke-sr**
issue: #18104

Signed-off-by: kejiang <ke.jiang@zilliz.com>

### CODE REVIEW

**Comment by aakejiang** [comment]
rerun ut

File: internal/indexcoord/index_builder.go [hunk 1]
```diff
@@ -154,6 +155,48 @@ func (ib *indexBuilder) schedule() {
...
+	// 1. populate: segment info from datacoord
+	segID := indexMeta.Req.SegmentID
+	segmentInfos, err := ib.ic.broker.getDataSegmentInfosByIDs([]UniqueID{segID})
+	if err != nil {
+		log.Error("get SegmentInfo failed", zap.Error(err))
+		return nil, fmt.Errorf("get SegmentInfo failed for segmentID %d", segID)
+	}
+
+	// 2. populate: field schema from rootcoord
+	collID := indexMeta.Req.CollectionID
```
Comment (xiaofan-luan) [inline internal/indexcoord/index_builder.go:168]: for old req, there are no collectionID and FieldIDs, it can not be compatible

Comment (aakejiang) [reply internal/indexcoord/index_builder.go:169]: fixed

**Comment by aakejiang** [comment]
For tasks, it needs saving snapshot fieldSchema & binlog paths for supporting time-travel.

**Pull request closed without merging**